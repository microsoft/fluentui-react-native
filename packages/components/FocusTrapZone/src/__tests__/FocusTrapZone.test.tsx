import * as React from 'react';

import type { IFocusable } from '@fluentui-react-native/interactive-hooks';
import * as renderer from 'react-test-renderer';

import { FocusTrapZone } from '..';

it('FocusTrapZone default props', () => {
  const tree = renderer.create(<FocusTrapZone />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('FocusTrapZone all props props', () => {
  const ref: React.RefObject<IFocusable> = { current: null };
  const tree = renderer
    .create(<FocusTrapZone componentRef={ref} disableFirstFocus disabled ignoreExternalFocusing focusPreviouslyFocusedInnerElement />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
