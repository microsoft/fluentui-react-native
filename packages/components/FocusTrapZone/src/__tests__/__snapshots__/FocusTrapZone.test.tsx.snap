// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusTrapZone all props props 1`] = `
<RCTFocusTrapZone
  disableFirstFocus={true}
  disabled={true}
  focusPreviouslyFocusedInnerElement={true}
  ignoreExternalFocusing={true}
/>
`;

exports[`FocusTrapZone default props 1`] = `<RCTFocusTrapZone />`;
